include/sys/bus.h

```diff
@@ -135,8 +136,9 @@ extern bus_space_t *generic_bus_space;
 #define bus_space_map(t, a, s, hp) (*(t)->bs_map)((a), (s), (hp))
 
 struct bus_methods {
-  void (*intr_setup)(device_t *dev, unsigned num, intr_handler_t *handler);
-  void (*intr_teardown)(device_t *dev, intr_handler_t *handler);
+  void (*intr_setup)(device_t *dev, resource_t *irq, ih_filter_t *filter,
+                     ih_service_t *service, void *arg, const char *name);
+  void (*intr_teardown)(device_t *dev, resource_t *irq);
```

Comment:
Maybe we should consider releasing an interrupt event in `intr_teardown` if we are removing the last interrupt handler?

Reply:
On the other hand, it will become problematic when we'll introduce an ithread per interrupt event.

Reply:
The right place to free `intr_event` would be in `driver_detach`. Since we don't have the method we don't have effective method to free interrupt events. OTOH in timer drivers we frequently register and unregister the only handler. That would needlessly create and destroy `intr_event` possibly along with interrupt thread. Let's not touch it right know. I'll leave a note in a comment.

Reply:
Right, I agree.